crates/gpui/src/platform/linux/x11/client.rs

```diff
@@ -143,6 +148,31 @@ impl X11Client {
             xkbc::x11::state_new_from_device(&xkb_keymap, &xcb_connection, xkb_device_id)
         };
 
+        let screen = xcb_connection.setup().roots.get(x_root_index).unwrap();
+
+        // Values from `Database::GET_RESOURCE_DATABASE`
+        let resource_manager = xcb_connection
+            .get_property(
+                false,
+                screen.root,
+                xproto::AtomEnum::RESOURCE_MANAGER,
+                xproto::AtomEnum::STRING,
+                0,
+                100_000_000,
+            )
+            .unwrap();
+        let resource_manager = resource_manager.reply().unwrap();
+
+        // todo(linux): read hostname
+        let resource_database = Database::new_from_default(&resource_manager, "HOSTNAME".into());
```

Comment:
Reading the hostname would require pulling extra crates, so I decided not to do it. It is only used as a fallback.